test/e2e/e2e_test.go

```diff
@@ -129,8 +129,39 @@ func TestE2EFlow(t *testing.T) {
 
 			// then
 			require.NoError(t, err)
-			verifyResourcesProvisionedForSignup(t, awaitility, johnSignup, revisions)
-			verifyResourcesProvisionedForSignup(t, awaitility, johnExtraSignup, revisions)
+			verifyResourcesProvisionedForSignup(t, awaitility, johnSignup, revisions, "basic")
+			verifyResourcesProvisionedForSignup(t, awaitility, johnExtraSignup, revisions, "basic")
+		})
+	})
+
+	t.Run("promote to advanced tier", func(t *testing.T) {
+		// given
+		hostAwaitility := wait.NewHostAwaitility(awaitility)
+		mur, err := hostAwaitility.WaitForMasterUserRecord(johnsmithName)
+		require.NoError(t, err)
+		mur.Spec.UserAccounts[0].Spec.NSTemplateSet.TierName = "advanced"
+
+		// when
+		err = awaitility.Client.Update(context.TODO(), mur)
+
+		// then
+		require.NoError(t, err)
+		verifyResourcesProvisionedForSignup(t, awaitility, johnSignup, revisions, "advanced")
+		verifyResourcesProvisionedForSignup(t, awaitility, johnExtraSignup, revisions, "basic")
```

Comment:
What if we introduce your "team" tier first and then test switching from "basic" to "team" and then back to "basic" with verifying that redundant namespaces are gone/re-stored? "advanced" and "basic" tiers are almost identical so I was thinking if testing "team" tier would be more interesting here.

Reply:
my idea was to merge these promotion PRs first and then modify this test together with introducing the team tier. 
I guess that both options should work.

Reply:
Yes, both option would work.